### PR TITLE
Adapt to SpringBoot 3, Camel 4 and Jakarta EE, removing DBCP feature and deprecated methods

### LINKED ISSUE #95: Support for Spring Boot 3
Hi,
looks current version of the lib is compatible with Spring Boot 2.X:

`<spring-boot.version>2.7.2</spring-boot.version>`

Would be good to support SpringBoot 3, which has been released recently.

### LINKED ISSUE #97: Spring Boot 3.0 support
Is there Any road map about spring boot 3.0 support. Transaction packages are change javax to jakarta

### PR DESCRIPTION
**Pull request by graben**
Major upgrade to latest Spring-Boot, Narayana and several other dependencies, replacing incompatible [DBCP ](https://github.com/apache/commons-dbcp) feature with [Agroal](https://github.com/agroal/agroal).

Resolves #95
Resolves #97 
Depends: agroal/agroal#63, agroal/agroal#65, agroal/agroal#66, agroal/agroal#67, agroal/agroal#68

### CODE REVIEW

**Comment by zhfeng** [comment]
Thanks @graben and how `io.agroal:agroal-spring-boot-starter` can work with `narayana` ? I didn't see any test with `agroal-spring-boot`.

**Comment by zhfeng** [comment]
OK - if that is the case, I think we should disable `XADataSourceWrapper` when using `io.agroal:agroal-spring-boot-starter`. `agroal` could take care all of the transaction integration.

Please make sure the transaction crash recovery is still working when you are adding test with `agroal`.

**Comment by graben** [comment]
Well, `XADataSourceWrapper` is still useful for non pooled DataSources which are not handled by `agroal`. Pooling has been completely removed from sourcecode, yet. The code at agroal has been developed by some Narayana guys from RedHat, I think they know what they are doing. ;-)

**Comment by zhfeng** [comment]
When using with `agroal-spring-boot`, `XADataSourceWrapper` will still wrap the `DatsSource` creaing by `agroal-spring-boot`?

File: narayana-spring-boot-starter/src/main/java/me/snowdrop/boot/narayana/autoconfigure/NarayanaConfiguration.java [hunk 1]
```diff
@@ -184,7 +182,6 @@ private File getLogDir() {
     /**
      * Generic data source wrapper configuration.
      */
-    @ConditionalOnProperty(name = "narayana.dbcp.enabled", havingValue = "false", matchIfMissing = true)
```
Comment (zhfeng) [inline narayana-spring-boot-starter/src/main/java/me/snowdrop/boot/narayana/autoconfigure/NarayanaConfiguration.java:187]: If we use `AgroalDataSource`, it should disable `XADataSourceWrapper` here.

**Comment by zhfeng** [comment]
@graben Can I ask for taking the removing dbcp in a seperate commit ? Since there is a [discussion](https://lists.apache.org/thread/kzoox2hkb484gyj1z13rn42xko2bp29w) to support jakarta in dbcp, so it could be easy to bring dbcp back in the future.

Thanks a lot!

**Comment by graben** [comment]
@zhfeng , I can restore DBCP support at any time. IMHO the discussion at commons is never ending, that's why I took the step forward for an alternative way. Btw agroal is creating a `DataSource` instance, so no problem with rewrapping by `XADataSourceWrapper` and they use narayana-spring-boot for their tests.

**Comment by graben** [comment]
Since no XADataSource instance is created, there is no problem with the GenericXADataSourceWrapper. Only one XAResourceRecoveryHelper is registered. But I can double-check with debugger in my personnel prototype app.

**Comment by zhfeng** [comment]
Thanks @graben - LGTM!

**Comment by graben** [comment]
@zhfeng: Changes good to be merged? It seems that your Fuse colleges at RedHat are fine with the changes proposed!

**Review by zhfeng — APPROVED** [review]

**Comment by zhfeng** [comment]
Thanks @graben - It looks good to me!

**Comment by graben** [comment]
@jacobdotcosta : Could you pls review and merge?

**Comment by jacobdotcosta** [comment]
I don't think I'm the right person to review this PR, could you take a look at it @Sgitario , please?

**Comment by Sgitario** [comment]
> I don't think I'm the right person to review this PR, could you take a look at it @Sgitario , please?

I have never worked in this repository either @jacobdotcosta

**Comment by geoand** [comment]
It looks fine from a strictly Spring Boot perspective

**Comment by graben** [comment]
@geoand : Would you like to approve and merge? The number of repository owners is quite small! Same for my other PRs.

**Review by geoand — APPROVED** [review]

**Comment by johnpoth** [comment]
Is there a release planned soon containing these changes ? Thanks !

**Comment by geoand** [comment]
@jacobdotcosta ^

**Pull request merged**